modules/openthread/platform/misc.c

```diff
@@ -9,6 +9,22 @@
 #include <openthread/instance.h>
 #include <openthread/platform/misc.h>
 
+#if defined(CONFIG_OPENTHREAD_PLATFORM_BOOTLOADER_MODE_RETENTION)
+
+#include <zephyr/retention/bootmode.h>
+
+#elif defined(CONFIG_OPENTHREAD_PLATFORM_BOOTLOADER_MODE_GPIO)
+
+BUILD_ASSERT(DT_HAS_COMPAT_STATUS_OKAY(openthread_config), "...");
+BUILD_ASSERT(DT_NODE_HAS_PROP(DT_COMPAT_GET_ANY_STATUS_OKAY(openthread_config), bootloader_gpios), "...");
```

Comment:
The assert suggestions where an example, please add a meaningful error message :)

Reply:
didn't notice the error message :) , fixed